SDVModTest/UIElements/LocationOfTownsfolk.cs

```diff
@@ -25,77 +26,86 @@ class LocationOfTownsfolk : IDisposable
+                        { "WizardHouse", new KeyValuePair<int, int>(196, 352) },
+                        { "Trailer", new KeyValuePair<int, int>(780, 360) },
+                        { "Forest", new KeyValuePair<int, int>(80, 272) },
+                        { "Woods", new KeyValuePair<int, int>(100, 272) },
+                        { "WitchSwamp", new KeyValuePair<int, int>(0, 0) },
+                        { "ArchaeologyHouse", new KeyValuePair<int, int>(892, 416) },
+                        { "FishShop", new KeyValuePair<int, int>(844, 608) },
+                        { "Saloon", new KeyValuePair<int, int>(714, 354) },
+                        { "LeahHouse", new KeyValuePair<int, int>(452, 436) },
+                        { "Town", new KeyValuePair<int, int>(680, 360) },
+                        { "Mountain", new KeyValuePair<int, int>(762, 154) },
+                        { "BusStop", new KeyValuePair<int, int>(516, 224) },
+                        { "Railroad", new KeyValuePair<int, int>(644, 64) },
+                        { "SkullCave", new KeyValuePair<int, int>(0, 0) },
+                        { "BathHouse_WomensLocker", new KeyValuePair<int, int>(576, 60) },
+                        { "Beach", new KeyValuePair<int, int>(790, 550) },
+                        { "BathHouse_MensLocker", new KeyValuePair<int, int>(576, 60) },
+                        { "Mine", new KeyValuePair<int, int>(880, 100) },
+                        { "WitchHut", new KeyValuePair<int, int>(0, 0) },
+                        { "AnimalShop", new KeyValuePair<int, int>(420, 392) },
+                        { "SamHouse", new KeyValuePair<int, int>(612, 396) },
+                        { "WitchWarpCave", new KeyValuePair<int, int>(0, 0) },
+                        { "Club", new KeyValuePair<int, int>(60, 92) }
+                };
+
+        #endregion
+
+        public LocationOfTownsfolk(ModOptions modOptions, ModConfig config, IModHelper helper)
         {
             _helper = helper;
-            _options = options;
+            _townspeopleOptions = config.Townspeople;
+
+            _showNPCLocationsOnMap = modOptions.GetOptionWithIdentifier<ModOptionToggle>(OptionKeys.ShowLocationOfTownsPeople) ?? new ModOptionToggle(OptionKeys.ShowLocationOfTownsPeople, "Show townspeople on map");
+            _showNPCLocationsOnMap.ValueChanged += ToggleShowNPCLocationsOnMap;
+            modOptions.AddModOption(_showNPCLocationsOnMap);
+            ToggleShowNPCLocationsOnMap(_showNPCLocationsOnMap.identifier, _showNPCLocationsOnMap.IsOn);
         }
 
         public void Dispose()
         {
-            ToggleShowNPCLocationsOnMap(false);
+            ToggleShowNPCLocationsOnMap(OptionKeys.ShowLocationOfTownsPeople, false);
         }
 
-        public void ToggleShowNPCLocationsOnMap(bool showLocations)
+        public void ToggleShowNPCLocationsOnMap(string identifier, bool showLocations)
         {
+            if (identifier != OptionKeys.ShowLocationOfTownsPeople)

```

Comment:
If you won't accept anything other than this value, why allow someone to pass anything in for it in the first place?

Reply:
because it is a framework, and its built to allow for different styles. Some people may to differentiate between identifiers in an event handler